legal-api/src/legal_api/reports/report.py

```diff
@@ -114,7 +114,20 @@ def _get_report(self):
         if response.status_code != HTTPStatus.OK:
             return jsonify(message=str(response.content)), response.status_code
 
-        if account_id is not None:
+        create_document = account_id is not None
+        create_filing_types = [
+          'incorporationApplication',
+          'continuationIn',
+          'amalgamation',
+          'registration'
+        ]
+        if self._filing.filing_type in create_filing_types:
+            create_document = create_document and self._business.idenentifier
```

Comment:
error referencing the 'identifier' here so it will always be false. I think there should be some test cases added to secure this logic as it is really important and easy to miss the need for it

Reply:
Test case added and above fixed